src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -215,13 +215,14 @@ def _find_resources(self, node, find_slots, cores_per_slot, gpus_per_slot,
                     gpus.append(gpu_idx)
                 gpu_idx += 1
 
-            core_map = [cores]
-            gpu_map  = [[gpu] for gpu in gpus]
+            cores_per_rank = cores_per_slot // ranks_per_slot
+            core_map = [cores[i:i + cores_per_rank]
+                        for i in range(0, len(cores), cores_per_rank)]
 
             slots.append({'node_name': node_name,
                           'node_id'  : node_id,
                           'core_map' : core_map,
-                          'gpu_map'  : gpu_map,
+                          'gpu_map'  : gpus,
```

Comment:
I would prefer if we keep the 'map' semantics for the `gpu_map`, in that is maps a rank to a specific GPU.  That would simplify the changes in the executor code (I think it would not really need changes?).  With that I mean that we map multiple ranks to the same gpu.  So, instead of 
```py
cpu_map = [[0, 1], [2, 3], [4, 5], [6, 7]]
gpu_map = [0, 1]
```
we would have
```py
cpu_map = [[0, 1], [2, 3], [4, 5], [6, 7]]
gpu_map = [[0], [0], [1], [1]]
```
(the double nesting is needed for multiple gpus per rank).  That way we keep the route open for heterogeneous ranks, otherwise we would need to change the executor again later.  Is there a good reason to change that data structure and the executor?

Reply:
> That way we keep the route open for heterogeneous ranks, otherwise we would need to change the executor again later.

how it would change?

without double nesting, it actually says that all these gpus are per slot, thus if there are multiple ranks per slot, then all these ranks share all gpus, in your case you assign one gpu per rank within the slot, but in case it is

```python
cpu_map = [[0, 1], [2, 3], [4, 5], [6, 7]]
gpu_map = [[0, 1], [0, 1], [0, 1], [0, 1]]
```

which looks messy, that's why I though to follow `gpu_map = [0, 1]` and it is clear it is per slot (which is either 1 rank or multi-ranks)

Reply:
> which looks messy

Why does that look messy to you?  It is nicely symmetric, isn't it?  I don't see a reason to have different data structures for gpus and cpus (or any other discreet resources) really.

Assume you have a task description `gpus_per_rank = 1.5` (which is valid now), and we have 6 ranks.  That in the new notion would result in 3 slots, each allocating 3 gpus.  Each slot would then be something like:

```py
cpu_map = [[0], [1]]
gpu_map = [0, 1, 2]
```

where each rank uses 50% of the specified GPU.   But an equally valid schedule could be that the first rank gets gpu 0, second rank gets gpu 2, and gpu 1 is shared.  That would not be expressible above (ranks are uniform there), but would work with:

```py
cpu_map = [[0], [1]]
gpu_map = [[0, 1], [1, 2]]
```

Yes, I know, it's a  contrived example, but that is what I mean with non-uniform ranks.  I don't see why the current change would be required by this PR.

Reply:
agreed and updated accordingly